rust/agama-lib/src/software/http_client.rs

```diff
@@ -0,0 +1,63 @@
+use crate::software::model::SoftwareConfig;
+use crate::{base_http_client::BaseHTTPClient, error::ServiceError};
+use std::collections::HashMap;
+
+pub struct SoftwareHTTPClient {
+    client: BaseHTTPClient,
+}
+
+impl SoftwareHTTPClient {
+    pub fn new() -> Result<Self, ServiceError> {
+        Ok(Self {
```

Comment:
you can use here `new_with_base` like `Ok(Self.new_with_base(BaseHTTPClient::new()?))`

Reply:
I see your point... IMHO my way is still not enough duplication to need the indirection :)